data_pipeline/gmail_data/get_gmail_data.py

```diff
@@ -90,10 +105,12 @@ def iter_gmail_history(service: Resource, user_id: str, start_id: str) -> Iterab
         yield from response['history']
 
 
-def get_gmail_history_details(service: Resource, user_id: str,  start_id: str) -> pd.DataFrame:
-    imported_timestamp = get_current_timestamp()
-    df_temp = pd.DataFrame(iter_gmail_history(service, user_id, start_id))
+def get_dataframe_for_history_response(
+        history_response: dict,
```

Comment:
based on the signature of `iter_gmail_history`, this seems to be an iterable of dict (this is something `mypy` would catch). i.e. type `Iterable[dict]`. Name could then be `history_response_iterable` or `history_iterable` (the latter would be closed to the function name `iter_gmail_history`, which doesn't mention `response`)

(I'd also suggest to change the related name of the variable in `get_gmail_history_details` or related)